lib/ptz.js

```diff
@@ -177,17 +177,32 @@ Cam.prototype.getStatus = function(options, callback) {
 		this._envelopeFooter()
 	}, function(err, data, xml) {
 		if (!err) {
-			var res = linerase(data).getStatusResponse.PTZStatus;
-			var status = {
-				position: {
-					x: res.position.panTilt.$.x
-					, y: res.position.panTilt.$.y
-					, zoom: res.position.zoom.$.x
+			// Fix for TypeError: Cannot read property 'panTilt' of undefined #91
+			var res = linerase(data);
+			var status = {};
+			if ('getStatusResponse' in res) {
+				if ('PTZStatus' in res.getStatusResponse) {
+					if ('position' in res.getStatusResponse.PTZStatus) {
+						status.position = {};
+						if ('zoom' in res.getStatusResponse.PTZStatus.position) {
+							status.position.zoom = res.getStatusResponse.PTZStatus.position.zoom.$.x;
+						}
+						if ('panTilt' in res.getStatusResponse.PTZStatus.position) {
+							status.position.x = res.getStatusResponse.PTZStatus.position.panTilt.$.x;
+							status.position.y = res.getStatusResponse.PTZStatus.position.panTilt.$.y;
+						}
+					}
+					if ('moveStatus' in res.getStatusResponse.PTZStatus) {
+						status.moveStatus = res.getStatusResponse.PTZStatus.moveStatus;
+					}
+					if ('utcTime' in res.getStatusResponse.PTZStatus) {
+						status.utcTime = res.getStatusResponse.PTZStatus.utcTime;
+					}
+					if ('error' in res.getStatusResponse.PTZStatus) {
```

Comment:
Should we be doing an error callback if this condition is met?

Reply:
For the utcTime? No. From the spec:
UTC Time – Specifies the UTC time when this status was generated.

Also, in the spec there is no 'error' in PTZStatus. That was existing code that I converted but decided to leave it as someone had reason to put it there.

I hope to go through all the code, eventually, and convet it like this to avoid exceptions.
Also, we don't look for specific errors (fault codes) which should be in the err of the callback:
env:Sender, ter:InvalidArgVal,  ter:NoProfile
Env:Sender, ter:InvalidArgVal,  ter:NoPTZProfile
Env: Receiver, ter:ActionNotSupported, ter:PTZNotSupported
Hope to get all these cleaned up too.